### PR TITLE
PR: Fix rotary_embedding_llama sweep test with proper golden function

### PR DESCRIPTION
**Pull request by stevendae**
### Problem

The existing `rotary_embedding_llama_model_traced.py` sweep test had a **fundamentally broken golden function** that made test results meaningless:

```python
# Previous broken implementation
try:
    torch_output_tensor = ttnn.get_golden_function(ttnn.experimental.rotary_embedding_llama)(...)
except:
    # BROKEN: Falls back to returning input unchanged!
    torch_output_tensor = torch_input_tensor  # This always passes PCC check
```

Additionally:
- **cos/sin tensors were randomly generated** instead of using proper frequency computation
- **trans_mat was randomly generated** instead of using the exact required structure
- **Decode mode was completely untested** (filtered out by invalidate_vector)
- Tests would pass even with completely wrong TTNN output because the "golden" was just the input

### What's Changed

**1. Implemented proper golden function** (`apply_rotary_emb_golden`)
- Uses real-number rotation math: `[cos -sin; sin cos] @ [x_even; x_odd]`
- Handles TTNN's "doubled" cos/sin format correctly
- Based on validated implementation from `test_rotary_embedding_llama.py`

**2. Fixed tensor generation**
- cos/sin now computed using production code path (`compute_gather_cos_sin`)
- trans_mat generated with `get_rot_transformation_mat` (exact `[1,1,32,32]` structure)
- Supports LLaMA 3 RoPE parameters with configurable theta, scaling, and rope_type

**3. Added full decode mode support**
- Automatic mode detection via HEIGHT_SHARDED memory layout
- Proper sharding setup with `ttnn.create_sharded_memory_config`
- Interleaved-to-sharded tensor conversion for all inputs
- Decode-specific golden reference computation

**4. Code cleanup**
- Removed unused imports (`torch_random`, `partial`)
- Removed unused parameters, added `**_kwargs` for flexibility
- Simplified redundant code paths
- Updated documentation for both prefill and decode modes

**5. Increased PCC threshold**
- Changed from 0.99 to 0.999 (proper golden achieves high accuracy)

### Test Results

| Metric | Before | After |
|--------|--------|-------|
| Prefill tests | Unknown (false positives) | **160 PASS** |
| Decode tests | 0 (filtered out) | **20 PASS** |
| Sample tests | Unknown | **1 PASS** |
| **Total** | Unknown | **181 PASS (100%)** |
| PCC | ~1.0 (broken golden) | **> 0.9999** (real validation) |

### Key Implementation Details

**Prefill Mode** (INTERLEAVED memory):
- Shape: `[batch, n_heads, seq_len, head_dim]`
- Uses standard DRAM interleaved memory

**Decode Mode** (HEIGHT_SHARDED memory):
- Shape: `[1, batch, n_heads, head_dim]`
- Sharded across batch dimension on cores
- Requires transformation matrix repeated per core

### CODE REVIEW

File: tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py [hunk 1]
```diff
@@ -49,120 +96,420 @@
...
-    )
+        # Sample configuration - derive shapes from input_shape
+        # input_shape format: [batch, n_heads, seq_len, head_dim]
+        shape_a = list(input_shape)
+        batch, n_heads, seq_len, head_dim = shape_a
+        # Generate cos/sin cache shapes (cache can be larger than seq_len)
+        cache_size = max(seq_len, 1024)  # Use at least 1024 for cache
+        shape_b = [1, 1, cache_size, head_dim]  # cos cache
+        shape_c = [1, 1, cache_size, head_dim]  # sin cache
+        shape_d = [1, 1, 32, 32]  # trans_mat (fixed size)
```
[the remark this answers is not in the document]
Comment (stevendae) [reply tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py]: unused variable removed

File: tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py [hunk 2]
```diff
@@ -49,120 +96,420 @@
...
-        shape_b
-    )
+        # Sample configuration - derive shapes from input_shape
+        # input_shape format: [batch, n_heads, seq_len, head_dim]
+        shape_a = list(input_shape)
+        batch, n_heads, seq_len, head_dim = shape_a
+        # Generate cos/sin cache shapes (cache can be larger than seq_len)
+        cache_size = max(seq_len, 1024)  # Use at least 1024 for cache
+        shape_b = [1, 1, cache_size, head_dim]  # cos cache
+        shape_c = [1, 1, cache_size, head_dim]  # sin cache
```
[the remark this answers is not in the document]
Comment (stevendae) [reply tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py:352]: unused variable removed

File: tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py [hunk 3]
```diff
@@ -48,121 +100,524 @@
...
+        # Input: [1, batch, n_heads, head_dim]
+        x_even = torch_input_tensor[..., 0::2]  # [1, batch, n_heads, head_dim//2]
+        x_odd = torch_input_tensor[..., 1::2]
+
+        # Apply 2D rotation
+        cos_part = x_even * freqs_cos - x_odd * freqs_sin
+        sin_part = x_even * freqs_sin + x_odd * freqs_cos
+
+        # Interleave back to original format
+        torch_output_tensor = torch.stack([cos_part, sin_part], dim=-1).flatten(-2).to(torch.bfloat16)
```
[the remark this answers is not in the document]
Comment (stevendae) [reply tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py:466]: The GitHub Copilot observation is technically accurate about the difference but doesn't indicate a bug—it's testing a different (simpler) scenario than the unit test.

File: tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py [hunk 4]
```diff
@@ -24,8 +70,9 @@
 # Parameters provided to the test vector generator are defined here.
 parameters = {
     # Quick sample test with basic configurations for fast validation
+    # Shape format for prefill: [batch, n_heads, seq_len, head_dim]
     "model_traced_sample": {
-        "input_shape": [(1, 1, 32, 64)],  # Batch, seq, heads, head_dim (must be even for rotary)
+        "input_shape": [(1, 8, 128, 64)],  # batch=1, n_heads=8, seq_len=128, head_dim=64
```
[the remark this answers is not in the document]
Comment (stevendae) [reply tests/sweep_framework/sweeps/model_traced/rotary_embedding_llama_model_traced.py:77]: The sample test is one configuration among the full traced suite. The current comment adequately documents what the shape represents. Adding more documentation for a sample config seems excessive.

**Review by bbradelTT — APPROVED** [review]
